sigar_windows.go

```diff
@@ -117,8 +118,70 @@ func (self *Cpu) populateFromPdh(values []uint64) {
 	self.Irq = values[3]
 }
 
+// Only get the first mount point for each volume
+func getVolumePathName(volumeName []uint16) (string, error) {
+	charsLen := C.DWORD(C.MAX_PATH)
+	var chars []uint16
+	for {
+		chars = make([]uint16, charsLen)
+		if C.GetVolumePathNamesForVolumeNameW((*C.WCHAR)(&volumeName[0]), (*C.WCHAR)(&chars[0]), C.MAX_PATH, &charsLen) == C.TRUE {
+			break
+		}
+		errno := syscall.GetLastError()
+		err, ok := errno.(syscall.Errno)
+		if !ok || err != C.ERROR_MORE_DATA {
+			return "", errno
+		}
+	}
```

Comment:
I'm having a lot of trouble understanding the control flow here.  I don't get the purpose of the for loop.  If we're successful or get an error that isn't ERROR_MORE_DATA we return, but if we do get ERROR_MORE_DATA there's no resizing so wouldn't we get stuck in an infinite loop if we get a ERROR_MORE_DATA?  Or is it the case that getting ERROR_MORE_DATA is impossible because we're sizing at MAX_PATH?  But if that's the case the for loop seems unnecessary.